applications/GeoMechanicsApplication/tests/cpp_tests/custom_elements/test_interface_element.cpp

```diff
@@ -0,0 +1,1187 @@
+    element.GetGeometry()[5].FastGetSolutionStepValue(DISPLACEMENT) = array_1d<double, 3>{0.2, 0.5, 0.0};
+
+    // Act
+    Vector actual_right_hand_side;
+    Matrix actual_left_hand_side;
+    element.CalculateLocalSystem(actual_left_hand_side, actual_right_hand_side, dummy_process_info);
+
+    // Assert
+    auto expected_left_hand_side    = Matrix{ZeroMatrix{12, 12}};
+    expected_left_hand_side(0, 0)   = shear_stiffness * (1.0 / 6.0);
+    expected_left_hand_side(1, 1)   = normal_stiffness * (1.0 / 6.0);
+    expected_left_hand_side(2, 2)   = shear_stiffness * (1.0 / 6.0);
+    expected_left_hand_side(3, 3)   = normal_stiffness * (1.0 / 6.0);
+    expected_left_hand_side(4, 4)   = shear_stiffness * (2.0 / 3.0);
+    expected_left_hand_side(5, 5)   = normal_stiffness * (2.0 / 3.0);
+    expected_left_hand_side(6, 6)   = shear_stiffness * (1.0 / 6.0);
+    expected_left_hand_side(7, 7)   = normal_stiffness * (1.0 / 6.0);
+    expected_left_hand_side(8, 8)   = shear_stiffness * (1.0 / 6.0);
+    expected_left_hand_side(9, 9)   = normal_stiffness * (1.0 / 6.0);
+    expected_left_hand_side(10, 10) = shear_stiffness * (2.0 / 3.0);
+    expected_left_hand_side(11, 11) = normal_stiffness * (2.0 / 3.0);
+
+    expected_left_hand_side(0, 6)  = -shear_stiffness * (1.0 / 6.0);
+    expected_left_hand_side(1, 7)  = -normal_stiffness * (1.0 / 6.0);
+    expected_left_hand_side(2, 8)  = -shear_stiffness * (1.0 / 6.0);
+    expected_left_hand_side(3, 9)  = -normal_stiffness * (1.0 / 6.0);
+    expected_left_hand_side(4, 10) = -shear_stiffness * (2.0 / 3.0);
+    expected_left_hand_side(5, 11) = -normal_stiffness * (2.0 / 3.0);
+
+    expected_left_hand_side(6, 0)  = -shear_stiffness * (1.0 / 6.0);
+    expected_left_hand_side(7, 1)  = -normal_stiffness * (1.0 / 6.0);
+    expected_left_hand_side(8, 2)  = -shear_stiffness * (1.0 / 6.0);
+    expected_left_hand_side(9, 3)  = -normal_stiffness * (1.0 / 6.0);
+    expected_left_hand_side(10, 4) = -shear_stiffness * (2.0 / 3.0);
+    expected_left_hand_side(11, 5) = -normal_stiffness * (2.0 / 3.0);
+
+    KRATOS_EXPECT_MATRIX_RELATIVE_NEAR(actual_left_hand_side, expected_left_hand_side, Defaults::relative_tolerance)
+
+    auto expected_right_hand_side = Vector{ZeroVector{12}};
+    expected_right_hand_side <<= 0.33333333, 1.66666667, 0.33333333, 1.66666667, 1.33333333,
+        6.66666667, -0.33333333, -1.66666667, -0.33333333, -1.66666667, -1.33333333, -6.66666667;
+    KRATOS_EXPECT_VECTOR_RELATIVE_NEAR(actual_right_hand_side, expected_right_hand_side, Defaults::relative_tolerance)
+}
+
+KRATOS_TEST_CASE_IN_SUITE(TriangleInterfaceElement_CreatesInstanceWithGeometryInput, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    // Arrange
+    const InterfaceElement element(0, std::make_shared<LineInterfaceGeometry2D2Plus2Noded>(CreateNodes()),
+                                   std::make_unique<Line2DInterfaceStressState>());
```

Comment:
For me it would make sense to have the types of geometries consistent here (although I think it probably will work like this, but it's just a bit confusing in the test)
```suggestion
    // Arrange
    const InterfaceElement element(0, std::make_shared<TriangleInterfaceGeometry3D3Plus3Noded>(Create6NodesForTriangle()),
                                   std::make_unique<SurfaceInterfaceStressState>());
```

Reply:
That is a very good catch!